data/src/main/java/com/microsoft/azure/kusto/data/auth/ManagedIdentityTokenProvider.java

```diff
@@ -0,0 +1,40 @@
+package com.microsoft.azure.kusto.data.auth;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.ManagedIdentityCredential;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URISyntaxException;
+
+public class ManagedIdentityTokenProvider extends TokenProviderBase {
+    private final ManagedIdentityCredential managedIdentityCredential;
+    private TokenRequestContext tokenRequestContext;
+
+    public ManagedIdentityTokenProvider(String managedIdentityClientId, @NotNull String clusterUrl) throws URISyntaxException {

```

Comment:
Usually clusterUrl is the first param

Reply:
I don't think that's the case.

Reply:
That's commonly the case in other SDKs.
Not sure of Java

Reply:
Not in Java

Reply:
Saw below and was horrified.
Let's make this one conform

Reply:
What was horrifying? That clusterUrl isn't the 1st param?

Reply:
That it moves from place to place depending on your choice of auth method

Reply:
I believe the authorityId was always at the end if applicable, and clusterUrl was always before that.